File: src/model_v4/entities/SBS_SW_OPDV.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, PrimaryColumn } from "typeorm";
import { SBS_SW } from "./SBS_SW";


@Entity("SBS_SW_OPDV",{schema:"sbsdb"})
@Index("sbsswopdv_index1",["DATUM",])
@Index("FKA04F6DD33E4C6525",["sW_INDEX",])
export class SBS_SW_OPDV {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"SWOPDV_INDEX"
        })
    SWOPDV_INDEX:string;
        

    @Column("date",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"DATUM"
        })
    DATUM:string | null;
        

    @Column("longtext",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"DOKLINK"
        })
    DOKLINK:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"FREIGABE"
        })
    FREIGABE:string | null;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"RISIKOSTUFE"
        })
    RISIKOSTUFE:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"VERSION"
        })
    VERSION:string | null;
        

   
    @ManyToOne(type=>SBS_SW, SBS_SW=>SBS_SW.sBS_SW_OPDVs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'SW_INDEX'})
    sW_INDEX:SBS_SW | null;

}

File: SbsdbServer/sbsdbv4/model/SbsSwOpdv.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_SW_OPDV")]
    public partial class SbsSwOpdv
    {
        [Column("SWOPDV_INDEX", TypeName = "bigint(20)")]
        public long SwopdvIndex { get; set; }
        [Column("DATUM", TypeName = "date")]
        public DateTime? Datum { get; set; }
        [Column("DOKLINK", TypeName = "longtext")]
        public string Doklink { get; set; }
        [Column("FREIGABE", TypeName = "varchar(50)")]
        public string Freigabe { get; set; }
        [Required]
        [Column("RISIKOSTUFE", TypeName = "varchar(50)")]
        public string Risikostufe { get; set; }
        [Column("VERSION", TypeName = "varchar(50)")]
        public string Version { get; set; }
        [Column("SW_INDEX", TypeName = "bigint(20)")]
        public long? SwIndex { get; set; }

        [ForeignKey("SwIndex")]
        [InverseProperty("SbsSwOpdv")]
        public virtual SbsSw SwIndexNavigation { get; set; }
    }
}
